src/inject/index.ts

```diff
@@ -42,16 +42,18 @@ function onMessage({type, data}) {
 }
 
 // TODO: Use background page color scheme watcher when browser bugs fixed.
-const colorSchemeWatcher = watchForColorSchemeChange(({isDark}) => {
+watchForColorSchemeChange(({isDark}) => {
     logInfo('Media query was changed');
     chrome.runtime.sendMessage({type: 'color-scheme-change', data: {isDark}});
 });
 
-const port = chrome.runtime.connect({name: 'tab'});
-port.onMessage.addListener(onMessage);
-port.onDisconnect.addListener(() => {
-    logWarn('disconnect');
-    cleanDynamicThemeCache();
-    colorSchemeWatcher.disconnect();
+chrome.runtime.onMessage.addListener(onMessage);
+chrome.runtime.sendMessage({type: 'tab'});
+
+window.addEventListener('unload', () => {
+    chrome.runtime.sendMessage({type: 'tab-unload'});
 });
 
+// TODO: figure out if this needs to call
```

Comment:
The original code included:
```
port.onDisconnect.addListener(() => {
    logWarn('disconnect');
    cleanDynamicThemeCache();
    colorSchemeWatcher.disconnect();
});
```
As far as I see, this code would not run during normal extension operation, since extension background page never called `disconnect()`. This code probably could run when extension is disabled or uninstalled because then background page is torn down (and its ports are closed), but I'm not sure how useful that is.

Reply:
Makes sense. I'll look into whether it's possible to notify extension about incoming reload to clean cache before reload or clean cache from the last version right upon reload.

Reply:
I have added this feature back in: https://github.com/darkreader/darkreader/pull/6138/commits/3063c315d25e6a8eadf1d0ceefc81d6df08827cc